downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,228 @@
+
+    ```ballerina
+    const MESSAGE = "message";
+    const map<string> HTTP_OK = {httpCode: "200", message: "OK", [MESSAGE] : "BAD REQUEST"}; // Compilation error now.
+    ```
+
+- Fixed a bug in the configurable TOML syntax validation for the module structure. The error message thrown for an invalid TOML module structure is now improved to provide the variable name.
+
+    For example, consider a non-default module `foo.bar`, which contains the following configurable variables.
+
+    ```ballerina
+    configurable int intVar = ?;
+    configurable string stringVar = ?;
+    ```
+
+    If an invalid TOML structure is found for the module `foo.bar` in the `Config.toml` file, it throws an error with the following message.
+
+    ```
+    [Config.toml:(1:1,2:23)] invalid TOML structure found for module ’foo.bar’. with variable 'intVar'. Please provide the module name as '[foo.bar]'
+    [Config.toml:(1:1,2:23)] invalid TOML structure found for module ‘foo’.bar. with variable 'stringVar'. Please provide the module name as ‘[foo.bar]’
+    ```
+
+## Platform updates
+
+### Official support for generating GraalVM native executables
+
+Ballerina now officially supports generating GraalVM native executables and the language and the standard libraries are compatible with the GraalVM native executable generation. In addition to that, GraalVM incompatibility warnings will be printed for any incompatible modules in the application. To explore more on this support, see the [Build a GraalVM executable](/learn/build-a-graalvm-executable/).
+
+## Language updates
+
+### New features
+
+#### Introduction of the `group by` and `collect` clauses
+
+The language now supports the `group by` and `collect` clauses to perform aggregation-related operations. The `group by` clause is used to group a collection based on a `grouping-key`. The `grouping-key` will be unique for each group. The `collect` clause is used to group a collection into one group. 
+
+```ballerina
+import ballerina/io;
+public type CovidEntry record {|
+    string district;
+    string province;
+    decimal deaths;
+|};
+public final table<CovidEntry> covidTable = table [
+    {district: "Colombo", province: "Western", deaths: 21},
+    {district: "Kandy", province: "Central", deaths: 14},
+    {district: "Kaluthara", province: "Western", deaths: 18},
+    {district: "Jaffna", province: "North", deaths: 10}
+];
+public function main() {
+    record {|string province; decimal deaths;|}[] deathsByProvince = from var {province, deaths} in covidTable
+        group by province
+        select {province, deaths: sum(deaths)};
+    io:println(deathsByProvince);
+    decimal totalDeaths = from var {deaths} in covidTable
+            collect sum(deaths);
+    io:println(totalDeaths);
```

Comment:
I would add the output as comments.

cc @KavinduZoysa @gimantha

Reply:
Btw, can't we use a non-COVID example?

Reply:
@gimantha @KavinduZoysa

Reply:
As we have the COVID example used in the Learn guides also (e.g., https://ballerina.io/learn/work-with-data-using-queries-in-ballerina/#the-complete-code) will use this here and will add another example in the blog.

Reply:
The Guide will be re-written again with a different examples. @KavinduZoysa Shall we change the example here?

Reply:
All the Get Started guides are written based on the Covid sample. Better to update them all if we are changing one and may be we can prioritize it later.

Shall we change only the aggregation example for the release note and blog for the moment?

Reply:
https://github.com/ballerina-platform/ballerina-dev-website/pull/7263 Please find the new examples for group-by and collect.